js/modules/loki_snode_api.js

```diff
@@ -756,73 +756,122 @@ class LokiSnodeAPI {
-    const allResults = [];
-    let ciphertextHex = null;
+    const confirmedNodes = [];
 
-    while (!ciphertextHex) {
-      if (lnsNodes.length < 3) {
-        log.error('Not enough nodes for lns lookup');
-        return false;
-      }
+    // Promise is only resolved when a consensus is found
+    let cipherResolve;
+    const cipherPromise = () =>
+      new Promise(resolve => {
+        cipherResolve = resolve;
+      });
 
-      // extract 3 and make requests in parallel
-      const nodes = lnsNodes.splice(0, 3);
+    const decryptHex = async cipherHex => {
+      const ciphertext = new Uint8Array(StringView.hexToArrayBuffer(cipherHex));
 
-      // eslint-disable-next-line no-await-in-loop
-      const results = await Promise.all(
-        nodes.map(node => this._requestLnsMapping(node, nameHash))
-      );
+      const res = await window.decryptLnsEntry(lnsName, ciphertext);
+      const pubicKey = StringView.arrayBufferToHex(res);
 
-      results.forEach(res => {
-        if (
-          res &&
-          res.result &&
-          res.result.status === 'OK' &&
-          res.result.entries &&
-          res.result.entries.length > 0
-        ) {
-          allResults.push(results[0].result.entries[0].encrypted_value);
-        }
-      });
+      return pubicKey;
+    };
 
-      const [winner, count] = _.maxBy(
-        _.entries(_.countBy(allResults)),
-        x => x[1]
-      );
+    const fetchFromNode = async node => {
+      const res = await this._requestLnsMapping(node, nameHash);
 
-      if (count >= 3) {
-        // eslint-disable-next-lint prefer-destructuring
-        ciphertextHex = winner;
+      // Do validation
+      if (res && res.result && res.result.status === 'OK') {
+        const hasMapping = res.result.entries && res.result.entries.length > 0;
+
+        const resValue = hasMapping
+          ? res.result.entries[0].encrypted_value
+          : null;
+
+        confirmedNodes.push(resValue);
+
+        if (confirmedNodes.length >= numRequiredConfirms) {
+          if (ciphertextHex) {
+            // Result already found, dont worry
+            return;
+          }
+
+          const [winner, count] = _.maxBy(
+            _.entries(_.countBy(confirmedNodes)),
+            x => x[1]
+          );
+
+          if (count >= numRequiredConfirms) {
+            ciphertextHex = winner === String(null) ? null : winner;
+
+            // null represents no LNS mapping
+            if (ciphertextHex === null) {
+              error = { lnsMappingNotFound: window.i18n('lnsMappingNotFound') };
+            }
+
+            cipherResolve({ ciphertextHex });
+          }
+        }
       }
-    }
+    };
 
-    const ciphertext = new Uint8Array(
-      StringView.hexToArrayBuffer(ciphertextHex)
-    );
+    const nodes = lnsNodes.splice(0, numRequests);
 
-    const res = await window.decryptLnsEntry(lnsName, ciphertext);
+    // Start fetching from nodes
+    nodes.map(async node => fetchFromNode(node));
```

Comment:
```suggestion
    nodes.forEach(node => fetchFromNode(node));
```
Since you don't use the mapped values, `forEach` is a better function to use

Reply:
These need to be async. Please see https://stackoverflow.com/a/37576787

Reply:
Right but you're not using the result anywhere, you're just executing promises which a `forEach` does fine, it's only when you need the result of the call should you `map` and `Promise.all` the logic

Reply:
Yeah, you just want to trigger all fetches from every nodes there without caring about the results, or the order. You can use foreach for that, without any async